runtime/compiler/il/J9Node.cpp

```diff
@@ -129,6 +129,16 @@ J9::Node::~Node()
    _unionPropertyB = UnionPropertyB();
    }
 
+bool
+J9::Node::dontEliminateStores(bool isForLocalDeadStore)
+   {
+   if (self()->getFirstChild()->getOpCode().isBinaryCodedDecimalOp())
+      return true;
```

Comment:
Could we put a comment down explaining why this restriction is in place? Perhaps:

```
// Disallow store sinking of BCD operations in Java because such operations may be under a BCDCHK node, implying that
// an exception path may be taken at runtime to evaluate the tree. Sinking such potentially exceptional operations
// violates the assumption that all such operations are to be anchored on a BCDCHK treetop, which the codegen expects
// to be able to generate the exception handling fallback code.
```

Reply:
Done! 4d2bbf4755e1af5868ae991a6eb3ee597e622b10